staging/src/k8s.io/client-go/discovery/discovery_client.go

```diff
@@ -230,14 +230,9 @@ func (d *DiscoveryClient) downloadLegacy() (*metav1.APIGroupList, map[schema.Gro
 		Do(context.TODO()).
 		ContentType(&responseContentType).
 		Raw()
-	// Special error handling for 403 or 404 to be compatible with older v1.0 servers.
-	// Return empty group list to be merged with /apis.
-	if err != nil && !errors.IsNotFound(err) && !errors.IsForbidden(err) {
```

Comment:
does removing this toleration make the discovery client unusable against an API server which only serves non-core resources?

Reply:
Removing this toleration will make any error returned by `/api` for the "core" group, an error for discovery as a whole. So yes, the API server must always serve discovery for the "core" group if we adopt this change.  Same for an API Server that only serves the "core" group, and returns 404 for `/apis`.

Reply:
hmm... do aggregated API servers serve resources at /api? I wouldn't have expected they would

Reply:
Wouldn't it be more accurate for the API Server to return `200 OK` and empty Group/Versions when the server knows it is not serving resources for those Group/Versions (instead of 404).

Reply:
I'm not sure why the client would ever tolerate `403`.

Reply:
um... a 404 for /api is pretty consistent with "I'm not a server that knows about or serves /api endpoints"

Reply:
cc @deads2k who does more with aggregated servers than I do

Reply:
Aggregated apiservers I'm used to seeing return a 404 for /api because they don't serve any core groupversion.  This behavior is comparable to returning 404 for "/apis/unknowngroup", where the server doesn't recognize or provide unknowngroup.

I agree that 403 is an unexpected value for an aggregated apiserver to return for /api, but 404 is normal.

Reply:
I will update the PR to tolerate the `404`, but not the `403`. Please let me know if there are any qualms with this approach.

Reply:
> I will update the PR to tolerate the 404, but not the 403. Please let me know if there are any qualms with this approach.

Yeah, I think that's a safer step

As an aside, I wonder if it would make sense to make the aggregated discovery doc at /apis serve core API info so we only have to request /apis with aggregated discovery. Asked in slack about that at https://kubernetes.slack.com/archives/C0EG7JC6T/p1677041390205779?thread_ts=1676658718.504059&cid=C0EG7JC6T

Reply:
We considered serving discovery only to `/apis` when creating the aggregated discovery client, but ultimately decided against it. It would probably be the best, eventual course, but it would be a significant effort. I'm guessing we would do this in a separate PR.

Reply:
>Hrmm... doesn't seem like it would be difficult to make kube-apiserver include core APIs in /apis aggregated response

I like that